### PR TITLE
added troubleshooting to fix #275

### LINKED ISSUE #275: Add new cases for troubleshooting guide
During negative testing, several issues were reported on Notation CLI, see follows:

- https://github.com/notaryproject/notation/issues/699
- https://github.com/notaryproject/notation/issues/700
- https://github.com/notaryproject/notation/issues/701

After discussion in the community, these issues couldn't be fixed in Notation v1 release, but it is worth adding these issues to troubleshooting guide. The mitigation is the same for all three issues, that is, user can run the command with `--verbose` or `-v` flag to see details of the error, and which helps users to understand what is happening and what to do next.

### PR DESCRIPTION
**Pull request by zr-msft**
fixes #275 
workaround for 
- https://github.com/notaryproject/notation/issues/699
- https://github.com/notaryproject/notation/issues/700
- https://github.com/notaryproject/notation/issues/701

### CODE REVIEW

**Review by FeynmanZhou — DISMISSED** [review]
Here are good examples of using `-v` to troubleshoot different error messages. It's acceptable.

**Review by yizha1 — DISMISSED** [review]
LGTM

**Comment by yizha1** [comment]
@iamsamirzon @priteshbandi would you mind take a look at this PR?

**Review by iamsamirzon — APPROVED** [review]

File: content/en/docs/troubleshooting.md [hunk 1]
```diff
@@ -141,6 +141,47 @@ SHA1 Thumbprint: xxx
...
+WARN Signature sha256:11223344556677889900aabbccddeeff11223344556677889900aabbccddeeff failed verification with error: error while loading the trust store, error while reading certificates from "/home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt": open /home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt: permission denied
+Error: signature verification failed for all the signatures associated with localhost:5001/net-monitor@sha256:000999888777666555444333222111fffeeedddcccbbbbaaa000999888777666
+```
+
+The following example shows there are no certificates in the `mystore` trust store.
+
+```console
+$ notation verify $IMAGE -v
+...
+ERRO authenticity validation failed. Failure reason: error while loading the trust store, trust store "/home/yizha/.config/notation/truststore/x509/ca/mystore" has no x509 certificates
```
Comment (Two-Hearts) [inline content/en/docs/troubleshooting.md:179]: nit:
```suggestion
ERRO authenticity validation failed. Failure reason: error while loading the trust store, trust store "/home/USER/.config/notation/truststore/x509/ca/mystore" has no x509 certificates
```

**Review by Two-Hearts — COMMENTED** [review]
@zr-msft Could you also address issues mentioned in [issue 291](https://github.com/notaryproject/notaryproject.dev/issues/291)? Thanks. /cc: @yizha1 @FeynmanZhou

File: content/en/docs/troubleshooting.md (hunk 1 above)
Comment (zr-msft) [reply content/en/docs/troubleshooting.md]: nice catch, fixed

**Comment by zr-msft** [comment]
@Two-Hearts I feel that #291 is out of scope for this PR and should be handled separately.

Per this, it seems that @Roseline-Bassey is already working on it: https://github.com/notaryproject/notaryproject.dev/issues/291#issuecomment-1598470805

**Review by yizha1 — APPROVED** [review]
LGTM

File: content/en/docs/troubleshooting.md [hunk 2]
```diff
@@ -139,6 +139,47 @@ SHA1 Thumbprint: xxx
...
 
+## When I verify an artifact, I get the error 'signature verification failed for all the signatures associated with IMAGE'
+
+This may result from one of the following issues:
+
+- Not having a trust store configured or the trust store is not configured correctly.
+- Trust store is not readable or the certificates stored in the trust store are not readable
+- no certificates were stored in trust store
+
+Rerun the verify command with `--verbose` to get more details.
```
Comment (priteshbandi) [inline content/en/docs/troubleshooting.md:150]:
```suggestion
```suggestion
This is expected error message when all the signatures associated with $IMAGE are not trusted.
There are some situation where misconfiguration of trust store might lead to above error message which includes:
- Trust store directory or the certificates stored in the trust store are not readable
- Trust store doesn't contains any certificate.
To debug this issues rerun the verify command with `--verbose` flag to get more details.
```

File: content/en/docs/troubleshooting.md [hunk 3]
```diff
@@ -139,6 +139,47 @@ SHA1 Thumbprint: xxx
...
+- Not having a trust store configured or the trust store is not configured correctly.
+- Trust store is not readable or the certificates stored in the trust store are not readable
+- no certificates were stored in trust store
+
+Rerun the verify command with `--verbose` to get more details. 
+
+The following example shows the trust store `mystore` has not be configured correctly and can't be found:
+
+```console
+$ notation verify $IMAGE -v
```
Comment (priteshbandi) [inline content/en/docs/troubleshooting.md:155]:
```suggestion
$ notation verify $IMAGE --verbose
```

File: content/en/docs/troubleshooting.md [hunk 4]
```diff
@@ -139,6 +139,47 @@ SHA1 Thumbprint: xxx
...
+
+This may result from one of the following issues:
+
+- Not having a trust store configured or the trust store is not configured correctly.
+- Trust store is not readable or the certificates stored in the trust store are not readable
+- no certificates were stored in trust store
+
+Rerun the verify command with `--verbose` to get more details. 
+
+The following example shows the trust store `mystore` has not be configured correctly and can't be found:
```
Comment (priteshbandi) [inline content/en/docs/troubleshooting.md:152]:
```suggestion
The following example shows the usecase where the trust store `mystore` doesn't exists:
```

File: content/en/docs/troubleshooting.md [hunk 5]
```diff
@@ -139,6 +139,47 @@ SHA1 Thumbprint: xxx
...
+
+```console
+$ notation verify $IMAGE -v
+...
+ERRO authenticity validation failed. Failure reason: error while loading the trust store, "/home/USER/.config/notation/truststore/x509/ca/mystore" does not exist
+WARN Signature sha256:11223344556677889900aabbccddeeff11223344556677889900aabbccddeeff failed verification with error: error while loading the trust store, "/home/USER/.config/notation/truststore/x509/ca/mystore" does not exist
+Error: signature verification failed for all the signatures associated with localhost:5001/net-monitor@sha256:000999888777666555444333222111fffeeedddcccbbbbaaa000999888777666
+```
+
+The following example shows the certificate `mystore.crt` in the `mystore` trust store could not be accessed:
```
Comment (priteshbandi) [inline content/en/docs/troubleshooting.md:162]:
```suggestion
The following example shows the usecase where notation couldn't read the certificate `mystore.crt` stored in `mystore` trust store:
```

File: content/en/docs/troubleshooting.md [hunk 6]
```diff
@@ -139,6 +139,47 @@ SHA1 Thumbprint: xxx
...
+
+```console
+notation verify --v $IMAGE
```
Comment (priteshbandi) [inline content/en/docs/troubleshooting.md:165]:
```suggestion
notation verify $IMAGE --verbose
```

File: content/en/docs/troubleshooting.md [hunk 7]
```diff
@@ -139,6 +139,47 @@ SHA1 Thumbprint: xxx
...
+...
+ERRO authenticity validation failed. Failure reason: error while loading the trust store, error while reading certificates from "/home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt": open "/home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt": permission denied
+WARN Signature sha256:11223344556677889900aabbccddeeff11223344556677889900aabbccddeeff failed verification with error: error while loading the trust store, error while reading certificates from "/home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt": open /home/USER/.config/notation/truststore/x509/ca/mystore/mystore.crt: permission denied
+Error: signature verification failed for all the signatures associated with localhost:5001/net-monitor@sha256:000999888777666555444333222111fffeeedddcccbbbbaaa000999888777666
+```
+
+The following example shows there are no certificates in the `mystore` trust store.
```
Comment (priteshbandi) [inline content/en/docs/troubleshooting.md:172]:
```suggestion
The following example shows the usecase where there is no certificate in the `mystore` trust store.
```

File: content/en/docs/troubleshooting.md [hunk 8]
```diff
@@ -139,6 +139,47 @@ SHA1 Thumbprint: xxx
...
+
+```console
+$ notation verify $IMAGE -v
```
Comment (priteshbandi) [inline content/en/docs/troubleshooting.md:175]:
```suggestion
$ notation verify $IMAGE --verbose
```

File: content/en/docs/troubleshooting.md (hunk 3 above)
Comment (FeynmanZhou) [reply content/en/docs/troubleshooting.md:155]: I think it's also acceptable to use a simple alias of the flag as an example.

File: content/en/docs/troubleshooting.md (hunk 8 above)
Comment (zr-msft) [reply content/en/docs/troubleshooting.md]: fixed

File: content/en/docs/troubleshooting.md (hunk 7 above)
Comment (zr-msft) [reply content/en/docs/troubleshooting.md]: made a minor wording change to clarify

File: content/en/docs/troubleshooting.md (hunk 6 above)
Comment (zr-msft) [reply content/en/docs/troubleshooting.md:165]: fixed

File: content/en/docs/troubleshooting.md (hunk 5 above)
Comment (zr-msft) [reply content/en/docs/troubleshooting.md:162]: i think the current wording is fine as is

File: content/en/docs/troubleshooting.md (hunk 4 above)
Comment (zr-msft) [reply content/en/docs/troubleshooting.md]: made a minor fix to clarify

File: content/en/docs/troubleshooting.md (hunk 3 above)
Comment (zr-msft) [reply content/en/docs/troubleshooting.md]: fixed

File: content/en/docs/troubleshooting.md (hunk 2 above)
Comment (zr-msft) [reply content/en/docs/troubleshooting.md]: fixed

File: content/en/docs/troubleshooting.md [hunk 9]
```diff
@@ -139,6 +139,47 @@ SHA1 Thumbprint: xxx
 
 Check the `Subject` info in the output. If it is the identity that signs the artifact, you need to add the `Subject` info into `trustedIdentities`.
 
+## When I verify an artifact, I get the error 'signature verification failed for all the signatures associated with IMAGE'
```
[the remark this answers is not in the document]
Comment (zr-msft) [reply content/en/docs/troubleshooting.md]: fixed

**Comment by zr-msft** [comment]
thanks @priteshbandi 

I incorporated most of your fixes and left some wording as is

**Review by yizha1 — APPROVED** [review]
LGTM

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Pull request merged**